### PR TITLE
typo definition

### PR DESCRIPTION
**Pull request by brookeclouston**
Added command line support for user to be able to define what a "typo" is (number of characters that is "allowed" to be different between two names that are being compared). Test script had to be updated to account for these modifications. Some comments changed too.

### CODE REVIEW

File: TypoSearch.py [hunk 1]
```diff
@@ -115,29 +112,30 @@ def main():
                         required=True)
     parser.add_argument("-table", action="store", dest="table", help="Name of table to search in", required=True)
     parser.add_argument("-rank", action="store", dest="rank", help="Rank level to start searching at", required=True)
+    parser.add_argument("-typo", action = "store", dest = "typo",
```
Comment (kcranston) [inline TypoSearch.py]: can we call this argument typo_limit or something else more descriptive?

File: TypoSearch.py [hunk 2]
```diff
@@ -115,29 +112,30 @@ def main():
...
+                        help= "Maximum number of character DIFFERENCES between two names to be considered a typo, "
```
Comment (kcranston) [inline TypoSearch.py]: good help message!

File: TypoSearch.py [hunk 3]
```diff
@@ -115,29 +112,30 @@ def main():
...
+                              "EX: 'foo' and 'foos!' have a character difference of 2", required= True)
     parser.add_argument("--show", action="store_true", dest="show", help="Print Locality ID's to be deleted to screen")
```
Comment (kcranston) [inline TypoSearch.py]: is show specific to just localities, or to taxa as well?

Comment (kcranston) [inline TypoSearch.py]: is this specific to localities only, or does show also apply to taxa?

File: TypoSearch.py [hunk 4]
```diff
@@ -115,29 +112,30 @@ def main():
                         required=True)
     parser.add_argument("-table", action="store", dest="table", help="Name of table to search in", required=True)
     parser.add_argument("-rank", action="store", dest="rank", help="Rank level to start searching at", required=True)
```
Comment (kcranston) [inline TypoSearch.py]: is rank an integer or string? Help message could be more helpful.

File: TypoSearch.py (hunk 1 above)
Comment (brookeclouston) [reply TypoSearch.py]: This argument has been changed to typo_limit.

File: TypoSearch.py (hunk 3 above)
Comment (brookeclouston) [reply TypoSearch.py]: This applies to taxa as well, the set `-typo_limit` is what the typo limit will be for whatever table is passed in with `-table`.

File: TypoSearch.py (hunk 4 above)
Comment (brookeclouston) [reply TypoSearch.py]: `-rank` is an integer, help message has been updated.

File: TypoSearch.py (hunk 3 above)
Comment (kcranston) [reply TypoSearch.py]: My comment ended up on the wrong line - I was asking about `show`. The help text says "Print Locality ID's..." but does `show` work with other tables as well?

Comment (brookeclouston) [reply TypoSearch.py]: Yes, `-show` also works with the table passed in with `-table` (Also this script doesn't delete anything). This mistake has been updated in the help message.

**Pull request merged**